src/backend/distributed/planner/multi_logical_planner.c

```diff
@@ -170,20 +172,20 @@ MultiLogicalPlanCreate(Query *originalQuery, Query *queryTree,
 
 
 /*
- * FindNodeCheck finds a node for which the check function returns true.
+ * FindNodeWithFilter finds a node for which the choose function returns true.
  *
  * To call this function directly with an RTE, use:
- * range_table_walker(rte, FindNodeCheck, check, QTW_EXAMINE_RTES_BEFORE)
+ * range_table_walker(rte, FindNodeWithFilter, chooseFunc, QTW_EXAMINE_RTES_BEFORE)
  */
 bool
-FindNodeCheck(Node *node, bool (*check)(Node *))
+FindNodeWithFilter(Node *node, ChooseFunc chooseFunc)
```

Comment:
naming is a bit inconsistent. If we call this WithFilter, shouldn't it be a filterFunc? Maybe nodeFilterFunc?

Reply:
Ah, just read the PR description. Well, I don't think choose is that clear, maybe a bit worse than check.

Reply:
We can also go with check if we can't find a better alternative.
`FindNodeWithCheck` doesn't seem super clear, but it is better than `FindNodeCheck`.

Reply:
I think check has some precedent in postgres:

```
extern bool check_functions_in_node(Node *node, check_function_callback checker,
                                    void *context);
```

Maybe `FindNodeMatchingCheckFunction(Node *node, CheckNodeFunc checker)`?

Reply:
I think `FindNodeMatchingCheckFunction(Node *node, CheckNodeFunc checker)` is very clear